### PR TITLE
 Adding ability to run getters in the debugger.

### PR DESCRIPTION
**Pull request by gushuro**
We add the ability to run getters in the debugger. 
When sending a variable's children request, now we can specify which ones of the class getters we should call to obtain values.

On the webapp side, we use this api to get getters marked to be displayed in their comment attributes.

### CODE REVIEW

File: pxtsim/debugger.ts [hunk 1]
```diff
@@ -124,29 +124,53 @@ namespace pxsim {
...
             }
             return r
         }
 
-        return frameVars(v);
+        return frameVars(v, fields);
+    }
+
+    function evalGetter(fn: LabelFn, target: RefObject) {
+        let p: any = {
```
Comment (riknoll) [inline pxtsim/debugger.ts]: Let's add some comments (or maybe not use single character var names)

File: pxtlib/service.ts [hunk 2]
```diff
@@ -158,6 +158,7 @@ namespace ts.pxtc {
         advanced?: boolean;
         deprecated?: boolean;
         useEnumVal?: boolean; // for conversion from typescript to blocks with enumVal
+        callInDebugger?: boolean; // for getters, they will be invoked by the debugger.
```
Comment (mmoskal) [inline pxtlib/service.ts]: callFromDebugger maybe?

File: pxtsim/debugger.ts (hunk 1 above)
Comment (riknoll) [reply pxtsim/debugger.ts]: (this was my fault)

Comment (gushuro) [reply pxtsim/debugger.ts]: Sure!

File: pxtlib/service.ts (hunk 2 above)
Comment (gushuro) [reply pxtlib/service.ts]: Already merged it in https://github.com/Microsoft/pxt-common-packages/pull/762, we can change it later if we feel it's not clear enough.

**Pull request merged**